File: src/com/leetcode/Medium/ContiguousArray.java
package com.leetcode.Medium;

// 525. Contiguous Array
public class ContiguousArray {
    public int findMaxLength(int[] nums) {
        int max = 0, count = 0;
        for (int i = 0; i < nums.length; i += 2) {
            if (i + 1 >= nums.length)
                break;
            if (nums[i] == 0 && nums[i + 1] == 1) {
                count += 2;
            } else {
                max = Math.max(count, max);
            }
        }

        return Math.max(count, max);
    }
}

File: JavaScript/Medium/ContiguousArray.js
/**
 * 525. Contiguous Array
 * @param {number[]} nums
 * @return {number}
 */
const findMaxLength = function (nums) {
  let map = new Map();
  map.set(0, -1);

  let max = 0, count = 0;
  for (let i = 0; i < nums.length; i++) {
    count += nums[i] == 1 ? 1 : -1;
    if (map.has(count)) {
      max = Math.max(max, i - map.get(count));
    } else {
      map.set(count, i);
    }
  }
  return max;
};